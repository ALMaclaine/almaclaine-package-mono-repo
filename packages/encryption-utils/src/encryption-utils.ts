import {hash, compare} from 'bcrypt';

export async function encryptPassword(password: string, rounds: number = 64000) {
    return hash(password, rounds);
}

export async function comparePassword(providedPassword: string, actualPassword: string) {
    return compare(providedPassword, actualPassword);
}
