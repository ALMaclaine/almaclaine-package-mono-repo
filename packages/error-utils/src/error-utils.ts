export class ALMError extends Error {
    constructor(message: string, public readonly type: string) {
        super(message);
    }
}
