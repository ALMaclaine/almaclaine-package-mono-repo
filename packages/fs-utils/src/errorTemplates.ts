export const doesNotExistError = (path: string): string => `Provided path ${path} does not exist: ${path}.`;

export const alreadyExistsError = (path: string): string => `Provided file/directory already exists: ${path}.`;

export const isNotDirectoryError = (path: string): string => `Provided path ${path} is not a directory.`;

export const isDirectoryError = (path: string): string => `Provided path ${path} is a directory.`;

export const isNotJsonFilError = (path: string): string => `Provided path ${path} is not a json file. Extension must be .json`;

