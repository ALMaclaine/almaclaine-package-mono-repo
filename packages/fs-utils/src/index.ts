export * from './fs-utils';
export * from './errorTemplates';
