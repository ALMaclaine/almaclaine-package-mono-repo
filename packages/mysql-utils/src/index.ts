export * from './mysql-utils';
export type {ConnectionInfo} from './types';
