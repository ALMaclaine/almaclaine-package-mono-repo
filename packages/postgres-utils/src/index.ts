export * from './postgres-utils';
export * from './ConnectionManager';
