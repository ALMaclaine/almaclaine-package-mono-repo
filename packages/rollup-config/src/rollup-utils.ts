import {
    nodeResolve
} from '@rollup/plugin-node-resolve';
import {join} from 'path';
const camelcase = require('camelcase');
const builtinModules = [
    "assert",
    "async_hooks",
    "buffer",
    "child_process",
    "cluster",
    "console",
    "constants",
    "crypto",
    "dgram",
    "dns",
    "domain",
    "events",
    "fs",
    "http",
    "http2",
    "https",
    "inspector",
    "module",
    "net",
    "os",
    "path",
    "perf_hooks",
    "process",
    "punycode",
    "querystring",
    "readline",
    "repl",
    "stream",
    "string_decoder",
    "timers",
    "tls",
    "trace_events",
    "tty",
    "url",
    "util",
    "v8",
    "vm",
    "zlib"
];

const {name, dependencies, peerDependencies} = require(join(process.cwd(), './package.json'));

const external = Object.keys({...dependencies, ...peerDependencies}).concat(builtinModules);
const globals = external.reduce((a, c: string) => {
    a[c] = camelcase(c);
    return a;
}, {});

function generateOutput(file: string, name: string) {
    return {
        file: `dist/${file}.bundle.js`,
        format: 'cjs',
        name: camelcase(name),
        globals
    }
}

export const PackageConfig = {
    input: 'dist/index.js',
    plugins: [nodeResolve()],
    external,
    output: [generateOutput('index', name)]
};

export const ScriptConfig = {
    ...PackageConfig,
    input: 'dist/exec.js',
    output: [
        ...PackageConfig.output,
        generateOutput('exec', name)
    ]
}
