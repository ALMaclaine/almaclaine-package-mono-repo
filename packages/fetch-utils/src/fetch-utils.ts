const fetch = require('cross-fetch');
import {ALMError} from "@almaclaine/error-utils";

export const ErrorTypes = {
    FetchFailed: 'FetchFailed'
}

export class FetchFailedError extends ALMError {
    constructor(message: string) {
        super(message, ErrorTypes.FetchFailed);
    }
}

export async function almFetch(url: string, options: {[key: string]: any} = {}) {
    const res = await fetch(url, options);
    if(!res.ok) {
        throw new FetchFailedError(`Fetch failed with status code: ${res.status}`);
    }
    return res;
}
