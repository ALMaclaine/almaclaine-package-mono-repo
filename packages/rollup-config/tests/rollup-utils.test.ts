import {PackageConfig, ScriptConfig} from '../src';

describe('utils/rollup-utils', () => {
    it('PackageConfig Matches Snapshopt', () => {
        expect(PackageConfig).toMatchSnapshot();
    });

    it('ScriptConfig Matches Snapshopt', () => {
        expect(ScriptConfig).toMatchSnapshot();
    });
})
