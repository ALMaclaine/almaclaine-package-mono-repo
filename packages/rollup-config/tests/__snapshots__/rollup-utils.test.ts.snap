// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/rollup-utils PackageConfig Matches Snapshopt 1`] = `
Object {
  "external": Array [
    "@babel/core",
    "@babel/preset-env",
    "@rollup/plugin-babel",
    "@rollup/plugin-node-resolve",
    "builtin-modules",
    "camelcase",
    "rollup",
    "assert",
    "async_hooks",
    "buffer",
    "child_process",
    "cluster",
    "console",
    "constants",
    "crypto",
    "dgram",
    "dns",
    "domain",
    "events",
    "fs",
    "http",
    "http2",
    "https",
    "inspector",
    "module",
    "net",
    "os",
    "path",
    "perf_hooks",
    "process",
    "punycode",
    "querystring",
    "readline",
    "repl",
    "stream",
    "string_decoder",
    "timers",
    "tls",
    "trace_events",
    "tty",
    "url",
    "util",
    "v8",
    "vm",
    "zlib",
  ],
  "input": "dist/index.js",
  "output": Array [
    Object {
      "file": "dist/index.bundle.js",
      "format": "cjs",
      "globals": Object {
        "@babel/core": "@babel/core",
        "@babel/preset-env": "@babel/presetEnv",
        "@rollup/plugin-babel": "@rollup/pluginBabel",
        "@rollup/plugin-node-resolve": "@rollup/pluginNodeResolve",
        "assert": "assert",
        "async_hooks": "asyncHooks",
        "buffer": "buffer",
        "builtin-modules": "builtinModules",
        "camelcase": "camelcase",
        "child_process": "childProcess",
        "cluster": "cluster",
        "console": "console",
        "constants": "constants",
        "crypto": "crypto",
        "dgram": "dgram",
        "dns": "dns",
        "domain": "domain",
        "events": "events",
        "fs": "fs",
        "http": "http",
        "http2": "http2",
        "https": "https",
        "inspector": "inspector",
        "module": "module",
        "net": "net",
        "os": "os",
        "path": "path",
        "perf_hooks": "perfHooks",
        "process": "process",
        "punycode": "punycode",
        "querystring": "querystring",
        "readline": "readline",
        "repl": "repl",
        "rollup": "rollup",
        "stream": "stream",
        "string_decoder": "stringDecoder",
        "timers": "timers",
        "tls": "tls",
        "trace_events": "traceEvents",
        "tty": "tty",
        "url": "url",
        "util": "util",
        "v8": "v8",
        "vm": "vm",
        "zlib": "zlib",
      },
      "name": "@almaclaine/rollupConfig",
    },
  ],
  "plugins": Array [
    Object {
      "buildStart": [Function],
      "generateBundle": [Function],
      "getPackageInfoForId": [Function],
      "load": [Function],
      "name": "node-resolve",
      "resolveId": [Function],
    },
  ],
}
`;

exports[`utils/rollup-utils ScriptConfig Matches Snapshopt 1`] = `
Object {
  "external": Array [
    "@babel/core",
    "@babel/preset-env",
    "@rollup/plugin-babel",
    "@rollup/plugin-node-resolve",
    "builtin-modules",
    "camelcase",
    "rollup",
    "assert",
    "async_hooks",
    "buffer",
    "child_process",
    "cluster",
    "console",
    "constants",
    "crypto",
    "dgram",
    "dns",
    "domain",
    "events",
    "fs",
    "http",
    "http2",
    "https",
    "inspector",
    "module",
    "net",
    "os",
    "path",
    "perf_hooks",
    "process",
    "punycode",
    "querystring",
    "readline",
    "repl",
    "stream",
    "string_decoder",
    "timers",
    "tls",
    "trace_events",
    "tty",
    "url",
    "util",
    "v8",
    "vm",
    "zlib",
  ],
  "input": "dist/exec.js",
  "output": Array [
    Object {
      "file": "dist/index.bundle.js",
      "format": "cjs",
      "globals": Object {
        "@babel/core": "@babel/core",
        "@babel/preset-env": "@babel/presetEnv",
        "@rollup/plugin-babel": "@rollup/pluginBabel",
        "@rollup/plugin-node-resolve": "@rollup/pluginNodeResolve",
        "assert": "assert",
        "async_hooks": "asyncHooks",
        "buffer": "buffer",
        "builtin-modules": "builtinModules",
        "camelcase": "camelcase",
        "child_process": "childProcess",
        "cluster": "cluster",
        "console": "console",
        "constants": "constants",
        "crypto": "crypto",
        "dgram": "dgram",
        "dns": "dns",
        "domain": "domain",
        "events": "events",
        "fs": "fs",
        "http": "http",
        "http2": "http2",
        "https": "https",
        "inspector": "inspector",
        "module": "module",
        "net": "net",
        "os": "os",
        "path": "path",
        "perf_hooks": "perfHooks",
        "process": "process",
        "punycode": "punycode",
        "querystring": "querystring",
        "readline": "readline",
        "repl": "repl",
        "rollup": "rollup",
        "stream": "stream",
        "string_decoder": "stringDecoder",
        "timers": "timers",
        "tls": "tls",
        "trace_events": "traceEvents",
        "tty": "tty",
        "url": "url",
        "util": "util",
        "v8": "v8",
        "vm": "vm",
        "zlib": "zlib",
      },
      "name": "@almaclaine/rollupConfig",
    },
    Object {
      "file": "dist/exec.bundle.js",
      "format": "cjs",
      "globals": Object {
        "@babel/core": "@babel/core",
        "@babel/preset-env": "@babel/presetEnv",
        "@rollup/plugin-babel": "@rollup/pluginBabel",
        "@rollup/plugin-node-resolve": "@rollup/pluginNodeResolve",
        "assert": "assert",
        "async_hooks": "asyncHooks",
        "buffer": "buffer",
        "builtin-modules": "builtinModules",
        "camelcase": "camelcase",
        "child_process": "childProcess",
        "cluster": "cluster",
        "console": "console",
        "constants": "constants",
        "crypto": "crypto",
        "dgram": "dgram",
        "dns": "dns",
        "domain": "domain",
        "events": "events",
        "fs": "fs",
        "http": "http",
        "http2": "http2",
        "https": "https",
        "inspector": "inspector",
        "module": "module",
        "net": "net",
        "os": "os",
        "path": "path",
        "perf_hooks": "perfHooks",
        "process": "process",
        "punycode": "punycode",
        "querystring": "querystring",
        "readline": "readline",
        "repl": "repl",
        "rollup": "rollup",
        "stream": "stream",
        "string_decoder": "stringDecoder",
        "timers": "timers",
        "tls": "tls",
        "trace_events": "traceEvents",
        "tty": "tty",
        "url": "url",
        "util": "util",
        "v8": "v8",
        "vm": "vm",
        "zlib": "zlib",
      },
      "name": "@almaclaine/rollupConfig",
    },
  ],
  "plugins": Array [
    Object {
      "buildStart": [Function],
      "generateBundle": [Function],
      "getPackageInfoForId": [Function],
      "load": [Function],
      "name": "node-resolve",
      "resolveId": [Function],
    },
  ],
}
`;
